siriuspy/siriuspy/fofb/csdev.py

```diff
@@ -0,0 +1,298 @@
+"""Define PVs, contants and properties of High Level FOFB."""
+
+import os as _os
+
+from .. import csdev as _csdev
+from ..search import PSSearch as _PSSearch, MASearch as _MASearch, \
+    BPMSearch as _BPMSearch
+
+
+# --- Enumeration Types ---
+
+class ETypes(_csdev.ETypes):
+    """Local enumerate types."""
+
+    OPEN_CLOSED = ('Open', 'Closed')
+
+    GLOB_INDIV = ('Global', 'Individual')
+
+    MEAS_RMAT_CMD = ('Start', 'Stop', 'Reset')
+    MEAS_RMAT_MON = ('Idle', 'Measuring', 'Completed', 'Aborted')
+
+    STS_LBLS_CORR = (
+        'Connected', 'PwrStateOn', 'OpModeConfigured', 'AccFreezeConfigured',
+        'InvRespMatRowSynced', 'LoopGainSynced')
+    STS_LBLS_FOFBCTRL = (
+        'Connected', 'BPMIdsConfigured', 'NetSynced', 'RefOrbSynced',
+        'TimeFrameLenSynced', 'BPMLogTrigsConfigured')
+
+
+_et = ETypes  # syntactic sugar
+
+
+# --- Const class ---
+
+class HLFOFBConst(_csdev.Const):
+    """Const class defining High Level FOFB constants."""
+
+    MIN_SING_VAL = 0.2
+    TIKHONOV_REG_CONST = 0
+    SINGVALHW_THRS = 1e-14
+
+    CONV_UM_2_NM = 1e3
+    ACCGAIN_RESO = 2**-12
+
+    LoopState = _csdev.Const.register('LoopState', _et.OPEN_CLOSED)
+    GlobIndiv = _csdev.Const.register('GlobIndiv', _et.GLOB_INDIV)
+    UseRF = _csdev.Const.register('UseRF', _et.DSBL_ENBL)
+    MeasRespMatCmd = _csdev.Const.register('MeasRespMatCmd', _et.MEAS_RMAT_CMD)
+    MeasRespMatMon = _csdev.Const.register('MeasRespMatMon', _et.MEAS_RMAT_MON)
+
+    def __init__(self):
+        """Class constructor."""
+
+        # device names and nicknames
+        self.bpm_names = _BPMSearch.get_names({'sec': 'SI', 'dev': 'BPM'})
```

Comment:
I think we should not use public attributes like `self.bpm_names` and others. we should use private attributes such as `self._bpm_names`and access to tho them with property and setter decoratored functions. This way all class interface is explicit and one does not to look at the class method implementation (like `__init__`) so find out all class interface.